src/metabase/lib/drill_thru.cljc

```diff
@@ -78,6 +78,19 @@
    (for [dimension dimensions]
      (merge context dimension))))
 
+(mu/defn- context-with-dimensions-or-row-dimensions :- ::lib.schema.drill-thru/context
+  "Return an updated `context` with either the existing `dimensions` or dimensions constructed from the `row`."
+  [query                                       :- ::lib.schema/query
+   {:keys [column dimensions row] :as context} :- ::lib.schema.drill-thru/context]
+  ;; If no dimensions were provided but the underlying column comes from an aggregation, then construct the dimensions
+  ;; from the row data. This is needed in cases where the frontend normally provides dimensions, but will not if it
+  ;; cannot determine that the clicked table cell was from an "underlying" aggregation-sourced column. See, e.g. the
+  ;; comment in getTableCellClickedObject in table.js.
+  (let [row-dimensions (lib.drill-thru.common/dimensions-from-breakout-columns query column row)]
+    (if (and (empty? dimensions) (not-empty row-dimensions))
```

Comment:
The idiom is to call `(seq something)` if we only care about it not being empty.

We only use `not-empty` itself when we care about the whole value, rather than a boolean.